client.go

```diff
@@ -913,15 +931,27 @@ func (f *File) Read(b []byte) (int, error) {
 // maximise throughput for transferring the entire file (especially
 // over high latency links).
 func (f *File) WriteTo(w io.Writer) (int64, error) {
-	fi, err := f.c.Stat(f.path)
-	if err != nil {
-		return 0, err
+	var err error
+	fileSize := uint64(0)
+	if f.c.useFstat {
+		var fileStat *FileStat
+		if fileStat, err = f.c.fstat(f.handle); err != nil {
+			return 0, err
+		}
+		fileSize = uint64(fileStat.Size)
+
+	} else {
+		var fi os.FileInfo
+		if fi, err = f.c.Stat(f.path); err != nil {
```

Comment:
Same as above, it is far more unwieldy to use an initializing precondition here, rather than just using the short declaration `:=` operator.